### ISSUE TITLE
Add masked arrays

### ISSUE THREAD

**Issue by donno2048** [google/jax#8979]
Please:
Add masked arrays (`numpy.ma.array`) it's doesn't have to be even a new feature it can be done by simply adding `mask` and `fill_value` arguments to the existing arrays (`jax.numpy.array`)
- [x] Check for duplicate requests.

**Comment by mattjj** [comment]
Thanks for the feature request!

I think adapting everything to work with masked arrays might take quite a bit of work. This may remain out of scope for us for a while. There's a lot of other high-priority work on our plates.

Some of this functionality could likely be implemented as a separate library on top of JAX itself.

**Comment by donno2048** [comment]
Is this planned for any time in the future?
